velox/functions/prestosql/tests/ArrayHasDuplicatesTest.cpp

```diff
@@ -0,0 +1,172 @@
+
+namespace {
+
+// Class to test the array_has_duplicates operator.
+class ArrayHasDuplicatesTest : public FunctionBaseTest {
+ protected:
+  // Evaluate an expression.
+  void testExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    auto result =
+        evaluate<SimpleVector<bool>>(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+
+  // Execute test for bigint type.
+  void testBigint() {
+    auto array = makeNullableArrayVector<int64_t>({
+        {},
+        {1,
+         std::numeric_limits<int64_t>::min(),
+         std::numeric_limits<int64_t>::max()},
+        {std::nullopt},
+        {1, 2, 3},
+        {2, 1, 1, -2},
+        {1, 1, 1},
+        {-1, std::nullopt, -1, -1},
+        {std::nullopt, std::nullopt, std::nullopt},
+        {1, -2, -2, 8, -2, 4, 8, 1},
+        {std::numeric_limits<int64_t>::max(),
+         std::numeric_limits<int64_t>::max(),
+         1,
+         std::nullopt,
+         0,
+         1,
+         std::nullopt,
+         0},
+    });
+
+    auto expected = makeNullableFlatVector<bool>(
+        {false, false, false, false, true, true, true, true, true, true});
+
+    testExpr(expected, "array_has_duplicates(C0)", {array});
+  }
+};
+
+} // namespace
+
+// Test integer arrays.
+TEST_F(ArrayHasDuplicatesTest, integerArrays) {
+  testBigint();
+}
+
+// Test inline (short) strings.
+TEST_F(ArrayHasDuplicatesTest, inlineStringArrays) {
+  using S = StringView;
+
+  auto array = makeNullableArrayVector<StringView>({
+      {},
+      {S("")},
+      {std::nullopt},
+      {S("a"), S("b")},
+      {S("a"), std::nullopt, S("b")},
+      {S("a"), S("a")},
+      {S("b"), S("a"), S("b"), S("a"), S("a")},
+      {std::nullopt, std::nullopt},
+      {S("b"), std::nullopt, S("a"), S("a"), std::nullopt, S("b")},
+  });
+
+  auto expected = makeFlatVector<bool>(
+      {false, false, false, false, false, true, true, true, true});
+
+  testExpr(expected, "array_has_duplicates(C0)", {array});
+}
+
+// Test non-inline (> 12 character length) strings.
+TEST_F(ArrayHasDuplicatesTest, stringArrays) {
+  using S = StringView;
+
+  auto array = makeNullableArrayVector<StringView>({
+      {S("red shiny car ahead"), S("blue clear sky above")},
+      {S("blue clear sky above"),
+       S("yellow rose flowers"),
+       std::nullopt,
+       S("blue clear sky above"),
+       S("orange beautiful sunset")},
+      {
+          S("red shiny car ahead"),
+          std::nullopt,
+          S("purple is an elegant color"),
+          S("red shiny car ahead"),
+          S("green plants make us happy"),
+          S("purple is an elegant color"),
+          std::nullopt,
+          S("purple is an elegant color"),
+      },
+  });
+
+  auto expected = makeFlatVector<bool>({false, true, true});
+  testExpr(expected, "array_has_duplicates(C0)", {array});
+}
+
+TEST_F(ArrayHasDuplicatesTest, nonContiguousRows) {
```

Comment:
This test is redundant as this logic is processed by the framework.

Reply:
Fixed